### DISCUSSION TITLE
Filtering models in "Early Access"

### DISCUSSION THREAD

**Discussion by Terroosh** [BlafKing/sd-civitai-browser-plus#90] [Ideas]
Are there any options in the api to let you filter out models in early access? These models will always fail to download.

Here's an example to test: https://civitai.com/models/183253/okuni

**Comment by BlafKing** [comment]
The API itself doesn't have an option to exclude these entirely from the search results.
(as far as I can tell at least, it's hard to discern the API's full capabilities since the [docs](https://github.com/civitai/civitai/wiki/REST-API-Reference) are not maintained)

However, they do return a field in the API return that defines if a model is early access, I'll make it so that any early access versions are hidden, and if the model only contains early access versions, that the entire model will then be hidden. :)

**Comment by BlafKing** [comment]
I've introduced some code that checks if any model is currently in early access, there isn't any documentation on how the early access field in the API return works so I had to do some guessing so please let me know if it works well!

The API returns `"earlyAccessTimeFrame":` followed by a number for each version.
I assume this number stands for the amount of days that the model will be in early access, since the [Okuni](https://civitai.com/models/183253/okuni) model you sent returns `5`

I've made some code that adds the amount of early access days to the upload date and then it looks if that date has passed, if not then I assume that means the model is still in early access so it will be hidden! :)
